### PR TITLE
Refactor `Activation_Flag` class

### LINKED ISSUE #9339: Use `PluginDeactivationAware`  and `PluginActivationAware`  in `Activation_Flag` class
## Feature Description

Use `PluginDeactivationAware`  and `PluginActivationAware`  in `Activation_Flag` class instead of registering the actions manually.

### PR DESCRIPTION
**Pull request by spacedmonkey**
…lag class.

## Summary

Use PluginDeactivationAware and PluginActivationAware in Activation_Flag class instead of registering the actions manually.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9339

### CODE REVIEW

**Comment by swissspidy** [comment]
Good idea 👍

**Comment by swissspidy** [comment]
I do wonder whether we should just merge all the `Activation_Flag` code into `Activation_Notice` though. There's no value in having both be services.

**Comment by spacedmonkey** [comment]
> I do wonder whether we should just merge all the Activation_Flag code into Activation_Notice though. There's no value in having both be services.

I was thinking the same. I will make the change.

File: includes/Admin/Activation_Notice.php [hunk 1]
```diff
@@ -84,6 +84,33 @@ public function register() {
...
+	 *
+	 * @param bool $network_wide Whether the deactivation was done network-wide.
+	 *
+	 * @return void
+	 */
+	public function on_plugin_deactivation( $network_wide ) {
+		$this->delete_activation_flag( $network_wide );
+	}
+
+
```
Comment (swissspidy) [inline includes/Admin/Activation_Notice.php]:
```suggestion
```

**Review by swissspidy — APPROVED** [review]
One teeny tiny suggestion but otherwise looks good 👍 🚀

**Pull request merged**